### PR TITLE
Use safer mxGetPr method

### LINKED ISSUE #944: Mex controllers incorrectly assume that the map pointer is a double
`constructQPDataPointerMex` and `QPControllermex` both call  

```
memcpy(&pdata->map_ptr,mxGetPr(prhs[6]),sizeof(pdata->map_ptr));
```

Changing `mxGetPr` to `mxGetPrSafe` reveals that the map ptr is actually a uint64, not a double, and we've just been getting lucky that they're the same size. Ick. 

I assume we want pointers to be stored as uint64 (or uint32 as appropriate) not doubles, right?

### PR DESCRIPTION
**Pull request by rdeits**
`mxGetPr()` *assumes* that it's been called on an `mxArray*` which corresponds to a Matlab double. If it's not called on a Matlab double (because, for example, you've accidentally passed in an `int` or a `single` or a `logical`), then it will just silently return a pointer to garbage data. 

The fix for this is easy: just check `mxIsDouble()` before you call `mxGetPr()`. To that end, I created `mxGetPrSafe` in `drakeUtil.cpp` which does just that (and throws a helpful error if you try to call it on something which is not a double).  This pull request changes every instance of `mxGetPr` to `mxGetPrSafe`, except in files which did not include `drakeUtil.hpp` yet (so as not to introduce any new linking dependencies). It also includes the changes from #945 

closes #944

### CODE REVIEW

**Comment by rdeits** [comment]
By the way, the code for mxGetPrSafe is here:

```
double * mxGetPrSafe(const mxArray *pobj) {
  if (!mxIsDouble(pobj)) mexErrMsgIdAndTxt("Drake:mxGetPrSafe:BadInputs", "mxGetPr can only be called on arguments which correspond to Matlab doubles");
  return mxGetPr(pobj);
}
```

**Comment by rdeits** [comment]
turns out `collisionDetect` was calling `mxGetPr` on a matrix of uint32s. I've changed it so as not to do that, so the next round of tests should have (a lot) fewer errors.

**Comment by rdeits** [comment]
Also, 196 unknown failures *has* to be some kind of record :wink:

File: examples/Atlas/+atlasControllers/@InstantaneousQPController/InstantaneousQPController.m [hunk 1]
```diff
@@ -83,7 +83,16 @@
...
+        % Try to determine if we're on 64- or 32-bit matlab. From http://www.mathworks.com/matlabcentral/answers/96172#comment_270161
+        % We do this because constructQPDataPointerMex now (correctly, I think) checks that the terrain_map_ptr
+        % is actually the correct size for a pointer on this system, rather than just blindly casting from double.
+        [~, maxArraySize] = computer();
+        is64bit = maxArraySize > 2^31;
+        if is64bit
+          terrain_map_ptr = uint64(0);
+        else
+          terrain_map_ptr = uint32(0);
+        end
```
Comment (RussTedrake) [inline examples/Atlas/+atlasControllers/@InstantaneousQPController/InstantaneousQPController.m]: would pull the rest of this now if it didn't include these commits (which I started discussing), too

Comment (rdeits) [inline examples/Atlas/+atlasControllers/@InstantaneousQPController/InstantaneousQPController.m]: I'd rather get it all in; we either have to call `mxGetPr` on a value that's known not to be a double (which I'm not a fan of) or include this fix.

**Comment by rdeits** [comment]
Ok, we now have a nullPointer() function which returns the correct data type for the system, and which is now used in the controller, RBM, IKOptions, and the trajectory classes

**Pull request merged**